src/remote.c

```diff
@@ -278,19 +281,19 @@ static int create_internal(git_remote **out, git_repository *repo, const char *n
 	else
 		remote->download_tags = GIT_REMOTE_DOWNLOAD_TAGS_AUTO;
 
-
-	git_buf_dispose(&var);
-
+	if (config)
+		git_config_unlock(config, 1);
 	*out = remote;
 	error = 0;
 
 on_error:
 	if (error)
 		git_remote_free(remote);
 
-	git_config_free(config_ro);
+	git_config_free(config);
```

Comment:
Don't we have to unlock the config on error, too?

Reply:
Will fix, I was misusing the transaction config API (hence the CI failure). The transaction handles unlocking when it's freed.